### PR TITLE
PERF: styler uuid control and security

### PR DESCRIPTION
**Pull request by attack68**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

`Styler` uuid is randomly generated from 16bytes, or 128bit entropy, which is poorly formatted for data transmission over the web.
`Styler` uuid1 is super-ceded by uuid4 which is more network secure.

This PR addresses the two above items by switching uuid method, and then coding the default entropy to 5 characters (20bit) which should be more than sufficient to avoid HTML table collision on a webpage, but make substantial data transfer savings for large tables.

uuid length remains configurable.

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -159,7 +164,8 @@ def __init__(
         self.index = data.index
         self.columns = data.columns
 
-        self.uuid = uuid
+        self.uuid_len = uuid_len if uuid_len < 33 else 32
```
Comment (jreback) [inline pandas/io/formats/style.py:167]: you can just use min

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -159,7 +164,8 @@ def __init__(
...
+        self.uuid = (uuid or uuid4().hex[: self.uuid_len]) + "_"
```
Comment (jreback) [inline pandas/io/formats/style.py:168]: no replacing - with _ ? why is that

File: pandas/io/formats/style.py [hunk 3]
```diff
@@ -73,6 +73,9 @@ class Styler:
         List of {selector: (attr, value)} dicts; see Notes.
     uuid : str, default None
         A unique identifier to avoid CSS collisions; generated automatically.
+    uuid_len : int, default 5
```
Comment (jreback) [inline pandas/io/formats/style.py:76]: i guess but this seems like overkill. add a versionadded tag

File: pandas/io/formats/style.py (hunk 1 above)
Comment (jreback) [inline pandas/io/formats/style.py:167]: i guess this should raise if its not an integer and must be > 0; pls add a test for this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 2 above)
Comment (attack68) [reply pandas/io/formats/style.py:167]: hex representation of uuid is 32 hex char string. no "-" are included, so no replace needed.

underscore at the end is to improve HTML readability.

File: pandas/io/formats/style.py [hunk 4]
```diff
@@ -73,6 +73,12 @@ class Styler:
...
```
Comment (WillAyd) [inline pandas/io/formats/style.py:76]: Minor but should append this to the end of the signature, in case people are calling positionally

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @WillAyd if you have any remaining comments.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @attack68

**Pull request merged**